solutions/cses-1132.mdx

```diff
@@ -80,7 +81,97 @@ int main() {
 	}
 	return 0;
 }
+```				 
+</CPPSection>
+	
+<JavaSection>
+	
+<Warning>
+Java exceeds the time limit on two test cases.
+</Warning>
+	
+```java
+
+import java.util.*;
+import java.io.*;
+
+public class TreeDistancesI {
+	static ArrayList<Integer>[] adj;
+	static int [] d1 = new int[200001];  // to store first-max length.
+	static int [] d2 = new int[200001]; // to store second-max length.
+	static int [] c = new int[200001]; // to store child for path of max length.
+
+	public static void dfs(int v, int p) {
+		d1[v] = 0;
+		d2[v] = 0;
+		for (int x : adj[v]) {
+			if (x == p) continue;
+			dfs(x, v);
+			if (d1[x] + 1 > d1[v]) {
+				d2[v] = d1[v];
+				d1[v] = d1[x] + 1;
+				c[v] = x;
+			}
+			else if (d1[x] + 1 > d2[v]) {
```

Comment:
```suggestion
			} else if (d1[x] + 1 > d2[v]) {
```

Reply:
this qualifies as an optional change

Reply:
Wait no it doesn't, it's not a personal preference. Oracle's styling guide dictates that you nest the if statements like above.

Reply:
also who the heck likes it that way anyways

Reply:
+1, this is taking formatting a bit too far